src/ApiResourceDataTable.php

```diff
@@ -0,0 +1,314 @@
+        $keyword = $this->config->isCaseInsensitive() ? Str::lower($keyword) : $keyword;
+
+        $this->collection = $this->collection->filter(function ($row) use ($keyword) {
+            $this->isFilterApplied = true;
+
+            $data = $this->serialize($row);
+            foreach ($this->request->searchableColumnIndex() as $index) {
+                $column = $this->getColumnName($index);
+                $value = Arr::get($data, $column);
+                if (! $value || is_array($value)) {
+                    if (! is_numeric($value)) {
+                        continue;
+                    }
+
+                    $value = (string) $value;
+                }
+
+                $value = $this->config->isCaseInsensitive() ? Str::lower($value) : $value;
+                if (Str::contains($value, $keyword)) {
+                    return true;
+                }
+            }
+
+            return false;
+        });
+    }
+
+    /**
+     * Perform default query orderBy clause.
+     */
+    protected function defaultOrdering()
+    {
+        $criteria = $this->request->orderableColumns();
+        if (! empty($criteria)) {
+            $sorter = $this->getSorter($criteria);
+
+            $this->collection = $this->collection
+                ->map(function ($data) {
+                    return array_dot($data);
+                })
+                ->sort($sorter)
+                ->map(function ($data) {
+                    foreach ($data as $key => $value) {
+                        unset($data[$key]);
+                        array_set($data, $key, $value);
+                    }
+
+                    return $data;
+                });
+        }
+    }
+
+    /**
+     * Get array sorter closure.
+     *
+     * @param array $criteria
+     * @return \Closure
+     */
+    protected function getSorter(array $criteria)
+    {
+        $sorter = function ($a, $b) use ($criteria) {
+            foreach ($criteria as $orderable) {
+                $column    = $this->getColumnName($orderable['column']);
+                $direction = $orderable['direction'];
+                if ($direction === 'desc') {
+                    $first  = $b;
+                    $second = $a;
+                } else {
+                    $first  = $a;
+                    $second = $b;
+                }
+                if ($this->config->isCaseInsensitive()) {
+                    $cmp = strnatcasecmp($first[$column], $second[$column]);
+                } else {
+                    $cmp = strnatcmp($first[$column], $second[$column]);
+                }
+                if ($cmp != 0) {
+                    return $cmp;
+                }
+            }
+
+            // all elements were equal
+            return 0;
+        };
+
+        return $sorter;
+    }
+
+    /**
+     * Resolve callback parameter instance.
+     *
+     * @return $this
+     */
+    protected function resolveCallbackParameter()
+    {
+        return $this;
+    }
+}
```

Comment:
I think this class can inherit from CollectionDataTable to avoid duplicate codes?